File: Behavioral/Strategy/JavaScript/paymentProcessor.js
// paymentProcessor.js

class PaymentProcessor {
  constructor() {
    this.paymentStrategy = null;
  }

  setPaymentStrategy(paymentStrategy) {
    this.paymentStrategy = paymentStrategy;
  }

  processPayment(amount) {
    if (!this.paymentStrategy) {
      console.log("No payment strategy selected.");
      return;
    }
    this.paymentStrategy.pay(amount);
  }
}

module.exports = PaymentProcessor;

File: Behavioral/Strategy/Cpp/PaymentProcessor.cpp
// PaymentProcessor.cpp

#include "PaymentProcessor.h"
#include <iostream>

PaymentProcessor::PaymentProcessor() : paymentStrategy(nullptr) {}

PaymentProcessor::~PaymentProcessor() {
  delete paymentStrategy;
}

void PaymentProcessor::setPaymentStrategy(PaymentStrategy* strategy) {
  delete paymentStrategy;
  paymentStrategy = strategy;
}

void PaymentProcessor::processPayment(double amount) {
  if (paymentStrategy == nullptr) {
    std::cout << "No payment strategy selected." << std::endl;
    return;
  }
  paymentStrategy->pay(amount);
}
